recipes/haproxy.rb

```diff
@@ -0,0 +1,35 @@
+#
+# Cookbook Name:: osl-jenkins
+# Recipe:: haproxy
+#
+# Copyright 2014, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+haproxy = node.default['haproxy']
+
+haproxy['enable_ssl'] = true
+haproxy['frontend_max_connections'] = '2000 \n' \
+  '  redirect scheme https if !{ ssl_fc }'
+haproxy['ssl_incoming_port'] = '443 ssl crt /etc/pki/tls/wildcard.pem ' \
+  'ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:' \
+  'ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS no-sslv3'
+haproxy['ssl_httpchk'] = false
+haproxy['members'] = [
+  {
+    'hostname' => 'jenkins',
+    'ipaddress' => '127.0.0.1',
+    'port' => '8080'
+  }
+]
```

Comment:
That what it should be doing with the line `haproxy =
node.default['haproxy']`

Will that not actually work? Do I need to explicitly write that everytime?

On Fri, Apr 3, 2015 at 1:47 PM, Anthony Miller <notifications@github.com>
wrote:

> In recipes/haproxy.rb
> <https://github.com/osuosl-cookbooks/osl-jenkins/pull/4#discussion_r27754779>
> :
>
> > +haproxy = node.default['haproxy']
> > +
> > +haproxy['enable_ssl'] = true
> > +haproxy['frontend_max_connections'] = '2000 \n' \
> > +  '  redirect scheme https if !{ ssl_fc }'
> > +haproxy['ssl_incoming_port'] = '443 ssl crt /etc/pki/tls/wildcard.pem ' \
> > +  'ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:' \
> > +  'ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS no-sslv3'
> > +haproxy['ssl_httpchk'] = false
> > +haproxy['members'] = [
> > +  {
> > +    'hostname' => 'jenkins',
> > +    'ipaddress' => '127.0.0.1',
> > +    'port' => '8080'
> > +  }
> > +]
>
> This actually doesn't change any attributes. I think you would prefer
> doing:
>
> node.default['haproxy'] = {
>   'enable_ssl' => 'foo',
>   'test' => 'bar'
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/osuosl-cookbooks/osl-jenkins/pull/4/files#r27754779>.
>